File: Retos/Reto #39 - TRIPLES PITAGÓRICOS [Media]/dart/JherysVargas.dart
import 'dart:math' as math;

void main() {
  final triplesFound = searchForPythagoreanTriples(10);
  print(triplesFound);
}

List<int> generateNumbers(int limit) {
  return List.generate(limit, (int index) => (index + 1));
}

List<List<int>> searchForPythagoreanTriples(int limit) {
  final List<List<int>> pythagoreanTriples = [];

  if (limit <= 0) return pythagoreanTriples;

  final List<int> numbers = generateNumbers(limit);

  for (int i = 0; i < numbers.length; i++) {
    for (int j = i + 1; j < numbers.length; j++) {
      final int a = numbers[i];
      final int b = numbers[j];
      final num cSquare = math.pow(a, 2) + math.pow(b, 2);

      final num c = math.sqrt(cSquare);

      if (c is int) {
        pythagoreanTriples.add([a, b, c.toInt()]);
      }
    }
  }

  return pythagoreanTriples;
}

File: Retos/Reto #39 - TRIPLES PITAGÓRICOS [Media]/typescript/JherysVargas.ts
const generateNumbers = (limit: number): number[] => {
  return Array.from({ length: limit }, (_, index) => index + 1);
};

const searchForPythagoreanTriples = (limit: number): number[][] => {
  const pythagoreanTriples: number[][] = [];

  if (limit <= 0) return pythagoreanTriples;

  const numbers: number[] = generateNumbers(limit);

  for (let i = 0; i < numbers.length; i++) {
    for (let j = i + 1; j < numbers.length; j++) {
      const a: number = numbers[i];
      const b: number = numbers[j];
      const cSquared: number = Math.pow(a, 2) + Math.pow(b, 2);
      const c: number = Math.sqrt(cSquared);

      if (Number.isInteger(c)) {
        pythagoreanTriples.push([a, b, c]);
      }
    }
  }

  return pythagoreanTriples;
};

const triplesFound: number[][] = searchForPythagoreanTriples(17);
console.log(triplesFound);
